### PR TITLE
Add `findall` method

### PR DESCRIPTION
**Pull request by CiaranOMara**
This PR:
- adds a docblock to the findall method.
- expands find examples where possible.
- adds doctests to unit tests.
- includes minor doctest corrections.
- enables `checkdocs` for build docs.

### CODE REVIEW

**Comment by CiaranOMara** [comment]
@SabrinaJaye, @jakobnissen, if these docs are good enough, could you make this the new HEAD?

**Comment by CiaranOMara** [comment]
@SabrinaJaye, could this be made the new HEAD?

File: src/BioSequences.jl [hunk 1]
```diff
@@ -227,4 +227,83 @@ include("search/ApproxSearchQuery.jl")
 include("search/re.jl")
 include("search/pwm.jl")
 
+
```
Comment (SabrinaJaye) [inline src/BioSequences.jl:230]: Can this be put in the search folder instead of the top level module file?

**Review by SabrinaJaye — CHANGES_REQUESTED** [review]

File: src/BioSequences.jl (hunk 1 above)
Comment (CiaranOMara) [reply src/BioSequences.jl]: I've put it into the search folder. Let me know if you had different else in mind.

File: src/search/search.jl [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+Find all occurrences of `pattern` in `sequence`.
+
+The return value is a vector of ranges of indices where the matching sequences were found.
+If there are no matching sequences, the return is an empty array.
+
+The search is restricted to the specified range when `rng` is set.
+
+With the keyword argument `overlap` set as `true`, the start index for the next search is set to the start of the current match plus one; if set to `false`, the start index for the next search is set to the end of the current match plus one.
+The default value for the keyword argument `overlap` is `true`.
```
Comment (jakobnissen) [inline src/search/search.jl:35]:
```suggestion
With the keyword argument `overlap` set as `true`, the start index for the next search is set to
the start of the current match plus one; if set to `false`, the start index for the next search is
set to the end of the current match plus one.
The default value for the keyword argument `overlap` is `true`.
```

File: src/search/search.jl [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+Base.eltype(::Type{<:Search}) = Int
+Base.IteratorSize(::Type{<:Search}) = Base.SizeUnknown()
+
+"""
+    findall(pattern, sequence::BioSequence[,rng::UnitRange{Int}]; overlap::Bool=true)
+
+Find all occurrences of `pattern` in `sequence`.
+
+The return value is a vector of ranges of indices where the matching sequences were found.
+If there are no matching sequences, the return is an empty array.
```
Comment (jakobnissen) [inline src/search/search.jl:30]:
```suggestion
    findall(pattern, sequence::BioSequence[,rng::UnitRange{Int}]; overlap::Bool=true)::Vector

Find all occurrences of `pattern` in `sequence`.

The return value is a vector of ranges of indices where the matching sequences were found.
If there are no matching sequences, return an empty array.

`pattern` can be a `Biosymbol` or a predicate.
```

File: test/longsequences/find.jl [hunk 4]
```diff
@@ -28,4 +28,26 @@
...
+    @test findall(ExactSearchQuery(dna"GATC", iscompatible), seq; overlap = true) == [1:4, 2:5, 6:9, 10:13]
+
+    # Check mapping of indices.
+    @test findall(DNA_A, seq, 7:11) == [7, 11]
+    @test findall(ExactSearchQuery(dna"A"), seq, 7:11) == [7:7, 11:11]
+
+    # Check empty return type.
+    @test findall(DNA_A, dna"GGGG") |> typeof == Vector{Int}
+    @test findall(ExactSearchQuery(dna"A"), dna"GGGG") |> typeof == Vector{UnitRange{Int}}
+
```
Comment (jakobnissen) [inline test/longsequences/find.jl:52]: Need tests for searching for predicate. See your comments in #235

File: src/search/search.jl [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+2-element Vector{UnitRange{Int64}}:
+ 1:4
+ 5:8
+
+julia> findall(ExactSearchQuery(dna"ACAC"), sequence, 2:7; overlap=false)
+1-element Vector{UnitRange{Int64}}:
+ 3:6
+```
+"""
+function Base.findall(pattern, sequence::BioSequence; overlap::Bool = DEFAULT_OVERLAP)
```
Comment (jakobnissen) [inline src/search/search.jl:68]: Ambiguous with `Base.findall(::Function, ::Any)` See #235

**Review by jakobnissen — CHANGES_REQUESTED** [review]
Thanks for this PR. Adding doctests is a good contribution, I see you found quite a lot of bugs.
I do have some slight wording issues in the docstring, and also the issues from #235 needs to be addressed.

**Comment by CiaranOMara** [comment]
Rebasing this PR was taking too long.

**Pull request closed without merging**